lib/ui/painting/codec.h

```diff
@@ -65,14 +67,22 @@ class MultiFrameCodec : public Codec {
   size_t decodedCacheSize_;
 
   std::vector<SkCodec::FrameInfo> frameInfos_;
-  // A struct linking the bitmap of a frame to whether it's required to render
-  // other dependent frames.
-  struct DecodedFrame {
-    std::unique_ptr<SkBitmap> bitmap_ = nullptr;
-    const bool required_;
-
-    DecodedFrame(bool required);
-    ~DecodedFrame();
+  // Handles logic based around caching required frames for future dependent
+  // frames.
+  class DecodedFrame {
+   public:
+    DecodedFrame(MultiFrameCodec* parent) : codec_(parent){};
+    const SkBitmap* ConsumeCachedBitmap();
+    bool CachedBitmapExists();
+    void MaybeCache(SkBitmap* bitmap);
```

Comment:
nit: I found "maybe" here a bit confusing before reading the implementation, though I cannot think of a good name ("CacheIfRequiredOrHasSpace"  is my best attempt 😄 )

I think a comment here would help.